go/obscuronode/enclave/enclave.go

```diff
@@ -392,16 +406,39 @@ func (e *enclaveImpl) GenerateSecret() obscurocommon.EncryptedSharedEnclaveSecre
 		panic(fmt.Sprintf("Could not generate secret: %s", err))
 	}
 	e.storage.StoreSecret(secret)
-	return encryptSecret(secret)
+	// todo do we need to worry about this failing?
+	encSec, _ := encryptSecret(e.publicKeySerialized, secret)
+	return encSec
 }
 
 // InitEnclave - initialise an enclave with a seed received by another enclave
-func (e *enclaveImpl) InitEnclave(secret obscurocommon.EncryptedSharedEnclaveSecret) {
-	e.storage.StoreSecret(decryptSecret(secret))
+func (e *enclaveImpl) InitEnclave(s obscurocommon.EncryptedSharedEnclaveSecret) {
+	secret, err := e.decryptSecret(s)
+	if err != nil {
+		panic("Failed to decrypt enclave secret.")
+	}
+	e.storage.StoreSecret(secret)
+}
+
+func (e *enclaveImpl) FetchSecret(att obscurocommon.AttestationReport) (obscurocommon.EncryptedSharedEnclaveSecret, error) {
+	data, err := e.attestationProvider.VerifyReport(att)
+	if err != nil {
+		return nil, err
+	}
+	if err = verifyIdentity(data, att); err != nil {
+		return nil, err
+	}
+	log.Log(fmt.Sprintf("Successfully verified attestation and identity. Owner: %v", att.Owner))
+	return encryptSecret(att.PubKey, e.storage.FetchSecret())
```

Comment:
`common.Address` can be printed with a `%s` as it implements the Stringer interface.